app/services/ui_notifications/hosts.rb

```diff
@@ -0,0 +1,23 @@
+module UINotifications
+  module Hosts
+    class Base < UINotifications::Base
+      def deliver!
+        if audience.nil?
+          raise(Foreman::Exception, "Invalid owner for #{subject}, unable to send notifications")
```

Comment:
do you expect to see this anywhere? i can change it to a log entry instead of raising, I was assuming this will be used by ActiveJob so it can halt the job, you are correct that currently with active record call back it might disable the action.